api/folders/controllers/folders.js

```diff
@@ -4,28 +4,51 @@ const { sanitizeEntity } = require("strapi-utils");
 
 module.exports = {
   async tree(ctx) {
-    let entities = await strapi.services.folders.find();
-    entities = entities.map((entity) =>
+    let foldersEntities = await strapi.services.folders.find();
+
+    foldersEntities = foldersEntities.map((entity) =>
       sanitizeEntity(entity, { model: strapi.models.folders })
     );
+
+    const { tag_id_in } = ctx.query;
+    let bookmarksFiltered = [];
```

Comment:
suggestion : `const bookmarksFilteredIds = new Set();`

Reply:
And then just push ids in it and remove the bookmarksFiltered.map (line 27)

Reply:
Héhé; bad old habbits

Reply:
Done